my-app/src/App.js

```diff
@@ -1,10 +1,52 @@
 import React, { Component } from 'react';
+import ListIterator from './ListIterator';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: []
+    }
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+
+  handleSubmit(event) {
+    let userInputElement = document.getElementById('userInput');
+    let currentUserInput = userInputElement.value;
+    if (currentUserInput !== '') {
+      if (this.state.items.indexOf(currentUserInput) > -1) {
+        alert('Duplicate Input');
+      } else {
+        this.state.items.push(currentUserInput);
```

Comment:
Never mutate state variables, Make a copy of it and modify copy then set back to the state using setState Method.
```
let items = [...this.state.items];
items.push(currentUserInput);
this.setState({
items: items
});
```